client/my-sites/controller.js

```diff
@@ -147,15 +148,19 @@ module.exports = {
 			if ( sites.fetched || ! sites.fetching ) {
 				return page.redirect( allSitesPath );
 			}
-			// Otherwise, check when sites has loaded
-			sites.once( 'change', function() {
+			onSitesChange = function() {
+				// if sites have not loaded yet, wait for next change
+				if ( ! sites.fetched ) return;
+				sites.off( 'change', onSitesChange );
```

Comment:
It should improve things because `sites` can emit multiple `'change'` events after data has been fetched from the server.